ecommerce/src/main/java/app/controller/CategoryController.java

```diff
@@ -0,0 +1,36 @@
+package app.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import app.authentication.MyDBAuthenticationService;
+import app.model.Category;
+import app.service.CategoryService;
+import app.service.Impl.CategoryServiceImpl;
+
+@Controller
+public class CategoryController {
+
+	@Autowired
+	private CategoryServiceImpl categoryService;
```

Comment:
```suggestion
	private CategoryService categoryService;
```
dùng interface chứ nhỉ